src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -142,8 +143,12 @@ static int AddItem(UCharList* list, const UChar item)
     size_t size = list->size++;
     if (size >= list->capacity)
     {
-        list->capacity *= 2;
-        UChar* ptr = (UChar*)realloc(list->items, list->capacity * sizeof(UChar*));
+        list->capacity += 512;
+        if (list->capacity < size)
```

Comment:
The PR is making a handful of unrelated improvements with the goal of improving perf... can you share measurements that demonstrate they move things in a positive direction?  I'm not clear on why this change to +512 instead of *2 is definitively better.

Reply:
The safety check I added is actually never going to get hit. GetCustomRules has upper bound on how many custom rules can get added, so realistically the only two cases that can happen is the default array size (512) or the next bigger one (1024 both before and after my change).

The maximum size at the moment is something around 860.

Reply:
I'll revert it to *2 since it doesn't really matter.